File: 0006-zigzag-conversion/0006-zigzag-conversion.js
/**
 * @param {string} s
 * @param {number} numRows
 * @return {string}
 */
var convert = function(s, numRows) {
     
    if(numRows == 1) return s;
    let arr  = new Array(numRows).fill("");
    
    let neg  = false;
    for(let i = 0,j=0; i < s.length; i++){
        
        if(!neg){
            arr[j++] += s[i];
        }else{
            arr[j--] += s[i];
        }
        
        if(j == numRows-1 || j == 0){
            neg = !neg;
        }
    }
    
    let str = "";
    for(let i = 0; i<numRows; i++){
        str += arr[i];
    }
    
    return str;
};

File: 0006-zigzag-conversion/0006-zigzag-conversion.cpp
class Solution {
public:
    string convert(string str, int r) {
        
        if(r == 1) return str;
        vector<string>ans(r,"");
        bool neg = false;
        for(int i = 0,j = 0; i<str.size(); i++){
            
            if(!neg){
                ans[j] += str[i];
                j++;
            }else{
                ans[j] += str[i];
                j--;
            }            
            
            if(j == r-1 || j == 0){
                neg = !neg;
            }
            
            // if(j >= r-1){
            //     j--;
            // }
            
        }
        
        string res;
        for(int i = 0; i<r; i++){
            cout << ans[i] << endl;
            res += ans[i];
        }
        
        return res;
    }
};